controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +121,241 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+						err := r.own(secret)
+						if err != nil {
+							return fmt.Errorf("failed to own Secret: %v", err)
+						}
+						secret.Data = data
+						return nil
+					})
+					if err != nil {
+						return reconcile.Result{}, fmt.Errorf("failed to create or update secret %v: %s", secret, err)
+					}
+				case "StorageClass":
+					var storageClass *storagev1.StorageClass
+					if resource.Name == "cephfs" {
+						if err != nil {
+							return reconcile.Result{}, fmt.Errorf("failed to enable cephfs csi: %v", err)
+						}
+						storageClass = r.getCephFSStorageClass(data)
+					} else if resource.Name == "ceph-rbd" {
+						storageClass = r.getCephRBDStorageClass(data)
+					}
+					err = r.createOrReplaceStorageClass(storageClass)
+					if err != nil {
+						return reconcile.Result{}, fmt.Errorf("failed to create or update StorageClass: %s", err)
+					}
+				}
+			}
+		}
+
+		// Readiness phase.
+		// Update the StorageClassClaim status.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimReady
+
+	}
+	// TODO(@rexagod): Deletion flow for SCC on Consumer side
+
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
+func (r *StorageClassClaimReconciler) getCephFSStorageClass(data map[string][]byte) *storagev1.StorageClass {
+	pvReclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.storageClassClaim.Name,
+			Namespace: r.storageClassClaim.Namespace,
+			Annotations: map[string]string{
+				"description": "Provides RWO and RWX Filesystem volumes",
+			},
+			Finalizers: []string{
+				StorageClassFinalizer,
+			},
+		},
+		ReclaimPolicy:        &pvReclaimPolicy,
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Provisioner:          fmt.Sprintf("%s.cephfs.csi.ceph.com", r.storageCluster.Namespace),
+	}
+	for k, v := range data {
+		storageClass.Parameters[k] = string(v)
+	}
+	return storageClass
+}
+
+func (r *StorageClassClaimReconciler) getCephRBDStorageClass(data map[string][]byte) *storagev1.StorageClass {
+	pvReclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.storageClassClaim.Name,
+			Namespace: r.storageClassClaim.Namespace,
+			Annotations: map[string]string{
+				"description": "Provides RWO Filesystem volumes, and RWO and RWX Block volumes",
+			},
+			Finalizers: []string{
+				StorageClassFinalizer,
+			},
+		},
+		ReclaimPolicy:        &pvReclaimPolicy,
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Provisioner:          fmt.Sprintf("%s.rbd.csi.ceph.com", r.storageCluster.Namespace),
+	}
+	for k, v := range data {
+		storageClass.Parameters[k] = string(v)
+	}
```

Comment:
No need to translate, as stated above data should be map[string]string in the first place which will allow you to just embed it in the structure above as Parameters: data